database/CoreDatabase.java

```diff
@@ -650,41 +660,45 @@ void deleted(CoreTransaction.Data transaction) {
          * Scan through all attributes that may need to be corrected (eg. have been over/under counted),
          * and correct them if we have enough information to do so.
          */
-        private void correctMiscounts() {
+        protected void correctMiscounts() {
```

Comment:
Split `correctMiscounts` into two methods - the "main" one and the "auxillary" one, allowing a subclass to override the "auxillary" method without having to duplicate the whole correction logic.

1. `correctMiscounts(CoreTransaction.Data txn)` is the "main" correction method, where the correction logic is defined
2. `correctMiscounts()` is the "auxillary" method, and it is responsible for instantiating a tx object and calling the "main" one.

Practically, this would allow Cluster to supply its own transaction + add extra logic for handling primary/secondary role transition that may happen concurrently with the miscount corrections.